frontend/src/web.py

```diff
@@ -120,6 +120,12 @@ def main():
     root.hosts = hostsfrontend.HostsFrontend()
     root.api = api.Root(root)   # JSON api exposure, enabling integration with other monitoring tools
 
+    if settings.get('oauth2',{}).get('switch'):
+        print 'switching on oauth2 ...'
+        import zalandoauth
+        root.zalandoauth = zalandoauth.ZalandOauth(settings)
```

Comment:
no need to pass all settings, if you only need settings['oauth2'], that'd also improve isolation of modules.